lib/ansible/plugins/lookup/consul_kv.py

```diff
@@ -33,9 +34,34 @@
            - The target to connect to, must be a resolvable address.
         env:
           - name: ANSIBLE_CONSUL_URL
+        ini:
+          - section: lookup_consul
+            key: host
       port:
         description: The port of the target host to connect to.
         default: 8500
+      scheme:
+        default: http
+        description: Whether to use http or https
+        version_added: '2.7'
```

Comment:
you describe scheme but give no way to set it?

Reply:
What do you mean by that? It can be set via the scheme param (line 131) or will be parsed from ANSIBLE_CONSUL_URL if one uses the environment variables

Reply:
@bcoca Can you please elaborate what the change request would be here? I still don't get it

Reply:
you have an option 'scheme' .. but the user has no way to set/change the option .. not via env var ini config .. anything

Reply:
@bcoca Ok, maybe i am overly tired or otherwise dense, but won't it be set via the kwargs?

For example ```lookup('consul_kv', 'my/key', host='10.10.10.10', port='2000', scheme='https')```

In line 131 i parse the kwargs for scheme
In line 125 i parse the ANSIBLE_CONSUL_URL From the environment. Both results will be used in the consul-library call (line 126 and 134).

Sorry in advance if i am mistaken and just don't see my failure here

Reply:
forgot that part, still it seems that rest of options can be set in other ways also, why force user to always set in kwargs

Reply:
@bcoca Because i didn't want to introduce backwards-compatibility breaking changes.
host and port were to be set via kwargs prior to my PR, so i thought it would be best to just stick to the scheme here.
If i removed it fom the kwargs and introduced explicit args for that, all usages of the lookup plugin would have to be rewritten correct?

Reply:
@bcoca i gave it a thought. In which way would you imagine that variable to be set in an other way than via the kwargs?

Reply:
env var or ini entry, just like you do for host

Reply:
@bcoca But it will already be parsed from ANSIBLE_CONSUL_URL. Shall i omit this parsed results **if** ANSIBLE_CONSUL_SCHEME is set **or always** and rely on ANSIBLE_CONSUL_SCHEME is set?